sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Hook/EmailHook.cs

```diff
@@ -19,24 +19,24 @@ public EmailHook(string name, IList<string> emailsToAlert)
         {
             Argument.AssertNotNull(emailsToAlert, nameof(emailsToAlert));
 
-            EmailsToAlert = emailsToAlert;
+            HookParameter = new EmailHookParameter(emailsToAlert);
             HookType = HookType.Email;
         }
 
         internal EmailHook(HookType hookType, string id, string name, string description, string externalLink, IReadOnlyList<string> administrators, EmailHookParameter hookParameter)
             : base(hookType, id, name, description, externalLink, administrators)
         {
-            EmailsToAlert = hookParameter.ToList;
+            HookParameter = hookParameter;
             HookType = hookType;
         }
 
         /// <summary>
         /// </summary>
-        public IList<string> EmailsToAlert { get; private set; }
+        public IList<string> EmailsToAlert => HookParameter.ToList;
```

Comment:
Could you explain to me the reason for the change? 
I feel like I am missing something. but, is it so the `EmailsToAlert` property only gets populated when it is explicitly asked for? while before it always had a value?

Reply:
Thanks for the explanation!